### DISCUSSION TITLE
How to get eth address from mintkey?

### DISCUSSION THREAD

**Discussion by yamijuan** [wormhole-foundation/wormhole#297] [Developer Q&A] [answered]
Hi,

What's the correct way to get the eth address of a token only having its corresponding solana mintkey part? I'm trying to use `.wrappedAssets(asset_id)` from the Wormhole contract but I have no Idea how to generate the asset id or address using the mint key.

**Comment by yamijuan** [comment]
btw, everything I've tried returns `0x0000000...`, I've encoded the chain and the mintkey (with padding to 32 and without) using `soliditySha3` but no luck

**Comment by yamijuan** [comment]
someone can help me with ths?

**Comment by yamijuan** [comment]
@leoluk could you help me to figure this out? I'm working to fix it in the Solana Wormholebridge project so this is important

**Comment by hendrikhofstadt** [comment, marked as answer]
@yamijuan https://github.com/certusone/wormhole/blob/db6a0738656622019fc5d89b75b0924b7c260693/web/src/utils/helpers.ts#L6-L12

**Comment by hendrikhofstadt** [comment]
Please note that Wormhole v2 is coming soon and we don't recommend using the v1 anymore.
